Practical Example/IosInterviewPracticeProject/IosInterviewPracticeProjectTests/CourseAPIServiceTests/CourseAPIServiceTests.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  NetworkManagerTests.swift
+//  IosInterviewPracticeProject
+//
+//  Created by Robert Ramirez on 8/20/20.
+//  Copyright © 2020 me.robert.ramirez. All rights reserved.
+//
+
+import XCTest
+@testable import raywenderlich
+
+class CourseAPIServiceTests: XCTestCase {
+  var artworkUrl: URL?
+  var courseAPIService = CourseAPIService()
+
+  override func setUpWithError() throws {
+    artworkUrl = URL(string: "https://files.betamax.raywenderlich.com/attachments/collections/244/e2d20345-06fb-42f4-8366-c7ffe80b5f77.png")
+  }
+
+    override func tearDownWithError() throws {
+      artworkUrl = nil
+    }
+
+  func testFetchArticles() {
+    let expect = expectation(description: "Fetch Articles")
+    let articleUrl = "https://raw.githubusercontent.com/raywenderlich/ios-interview/master/Practical%20Example/articles.json"
+    var items = [Item]()
+
+    courseAPIService.fetchContent(for: articleUrl) { results in
+      switch results {
+      case .success(let results):
+        items = results.data
+        expect.fulfill()
+      case .failure(let error):
+        XCTFail(error.localizedDescription)
+      }
+    }
+    waitForExpectations(timeout: 3, handler: nil)
+    XCTAssertGreaterThan(items.count, 0, "Failed to fetch articles")
+  }
+
+  func testFetchVideos() {
+    let expect = expectation(description: "Fetch Videos")
+    let videoUrl = "https://raw.githubusercontent.com/raywenderlich/ios-interview/master/Practical%20Example/videos.json"
+    var items = [Item]()
+
+    courseAPIService.fetchContent(for: videoUrl) { results in
+      switch results {
+      case .success(let results):
+        items = results.data
+        expect.fulfill()
+      case .failure(let error):
+        XCTFail(error.localizedDescription)
+      }
+    }
+    waitForExpectations(timeout: 3, handler: nil)
+    XCTAssertGreaterThan(items.count, 0, "Failed to fetch videos")
```

Comment:
When we disregard that the tests connect to the internet, your tests are well-written and structured and test the right things. To be more thorough with your tests, make sure to also test the other cases: What happens if the server returns an invalid JSON, what if there is an error etc. These cases are sometimes more important to test: You might not encounter these errors during development, but you need to know that your app won't crash if your server goes down, for instance.